flow/workflows/cdc_flow.go

```diff
@@ -502,27 +503,27 @@ func CDCFlowWorkflow(
 			var sleepFor time.Duration
 			var panicErr *temporal.PanicError
 			if errors.As(err, &panicErr) {
+				sleepFor = time.Duration(10+min(state.ErrorCount, 3)*15) * time.Minute
 				logger.Error(
-					"panic in sync flow, waiting 10 minutes",
+					"panic in sync flow",
 					slog.Any("error", panicErr.Error()),
 					slog.String("stack", panicErr.StackTrace()),
+					slog.Any("sleepFor", sleepFor),
 				)
-				sleepFor = 10 * time.Minute
 			} else {
-				logger.Error("error in sync flow", slog.Any("error", err))
-
 				// cannot use shared.IsSQLStateError because temporal serialize/deserialize
 				if !temporal.IsApplicationError(err) || strings.Contains(err.Error(), "(SQLSTATE 55006)") {
-					logger.Info("sync flow errored, waiting 1 minute before retrying")
-					sleepFor = time.Minute
+					sleepFor = time.Duration(1+min(state.ErrorCount, 9)) * time.Minute
 				} else {
-					logger.Info("sync flow errored, waiting 10 minutes before retrying")
-					sleepFor = 10 * time.Minute
+					sleepFor = time.Duration(5+min(state.ErrorCount, 5)*15) * time.Minute
 				}
+
+				logger.Error("error in sync flow", slog.Any("error", err), slog.Any("sleepFor", sleepFor))
 			}
 			mainLoopSelector.AddFuture(model.SleepFuture(ctx, sleepFor), func(_ workflow.Future) {
 				logger.Info("sync finished after waiting after error")
 				finished = true
+				finishedError = true
```

Comment:
maybe set this sooner so pause doesn't reset counter, but having counter reset on pause/resume seems like a feature

Reply:
+1, pause + resume should reset the counter.
In cases user is trying to fix the error, not having to wait to see if the fix worked would by resetting this externally would be valuable